src/utils/wallet.js

```diff
@@ -18,16 +20,37 @@ export function WalletProvider({ children }) {
     'https://www.sollet.io',
   );
 
-  const wallet = useMemo(() => new Wallet(providerUrl, endpoint), [
+  let wallet = useMemo(() => new Wallet(providerUrl, endpoint), [
```

Comment:
This part still has a problem of creating 2 wallet instances, and only 1 is used. Please check the condition, and create only Wallet or EzWallet.